### PR TITLE
[receiver/tlscheck] Inital Commit of TLS Check Receiver

### PR DESCRIPTION
**Pull request by michael-burt**
**Description:** Initial commit of TLS Check Receiver, a new component aimed at emitting metrics about the expoiry of x.509 certs.

**Link to tracking Issue:** [35423](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35423)

**Testing:** Test files added

**Documentation:** `README.md` & `metadata.yaml` added

### CODE REVIEW

File: receiver/tlscheckreceiver/README.md
Comment (crobert-1) [inline receiver/tlscheckreceiver/README.md:2]:
```suggestion

<!-- status autogenerated section -->
<!-- end autogenerated section -->

```
Adding this will allow `make generate` to generate the header that all components have in their README 👍

Comment (michael-burt) [reply receiver/tlscheckreceiver/README.md:2]: Thank you. I was wondering why none of the autogenerated code did not appear when I ran `make generate`

**Review by atoulme — APPROVED** [review]

File: receiver/tlscheckreceiver/README.md
Comment (breedx-splk) [inline receiver/tlscheckreceiver/README.md:19]: Is it worth saying explicitly that this value will be negative if the expiry time has been passed?

Comment (michael-burt) [reply receiver/tlscheckreceiver/README.md:19]: I was thinking it would clamp to 0 after expiry, but open to changing that behavior. I haven't often encountered negative values in gauges. I will document the behavior explicitly.

File: receiver/tlscheckreceiver/config.go
Comment (MovieStoreGuy) [inline receiver/tlscheckreceiver/config.go:58]: From memory, the `component.ValidateConfig(...)` method performs reflection instead of type inference so it can explore each field if they are set. 

It might be possible that your Targets are being evaluated twice, do you mind just double checking this isn't the case?

File: receiver/tlscheckreceiver/config_test.go
Comment (MovieStoreGuy) [inline receiver/tlscheckreceiver/config_test.go:27]: For future reference, `multierr.Combine` will return the original err if there is only one set, so the call the method here isn't needed.

File: receiver/tlscheckreceiver/metadata.yaml
Comment (MovieStoreGuy) [inline receiver/tlscheckreceiver/metadata.yaml:9]: Please also include @atoulme as a code owner since they sponsored this component.

**Review by MovieStoreGuy — APPROVED** [review]
Just a few minor nits (and to be honest, they don't need actioning beyond adding Antoine). 

Once that is added, I will be happy to merge the PR for you.

File: receiver/tlscheckreceiver/config.go
Comment (MovieStoreGuy) [reply receiver/tlscheckreceiver/config.go:58]: Looking at the test cases, it looks like it doesn't evaluate slices so I think it is fine to ignore this comment :)

File: receiver/tlscheckreceiver/metadata.yaml
Comment (michael-burt) [reply receiver/tlscheckreceiver/metadata.yaml]: fixed

File: receiver/tlscheckreceiver/config_test.go
Comment (michael-burt) [reply receiver/tlscheckreceiver/config_test.go]: fixed

**Comment by MovieStoreGuy** [comment]
Please take a look at the failing linting :)

File: receiver/tlscheckreceiver/README.md
Comment (breedx-splk) [reply receiver/tlscheckreceiver/README.md]: It would allow a user to know how long ago it expired, not just that it had expired.

Comment (michael-burt) [reply receiver/tlscheckreceiver/README.md]: I added it [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35441/files#diff-fd4a88524697321ef8eb94214702d17988629021b4f234aa751899645eefe5dcR19) and [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35441/files#diff-545eea690385fa2a051465cbef7a0487d9f6a9fc47721eef00f97f92487babc5R30).

File: receiver/tlscheckreceiver/config.go
Comment (dehaansa) [inline receiver/tlscheckreceiver/config.go:41]: What is the expected behavior (of validate && the receiver) for a non-https URL? I would expect it to be invalid at validate time, as I don't see a good mechanism for returning that information at runtime in a way that would be consumable.

Comment (MovieStoreGuy) [reply receiver/tlscheckreceiver/config.go:41]: That is a good point and something definitely overlooked. 

For the sake of the PR being open so long and becoming harder to merge, I would love to see this as a follow up PR instead updating the current PR.

Comment (michael-burt) [reply receiver/tlscheckreceiver/config.go:41]: I don't think we should allow for scheme. We do not need to send an http request here, only establish tcp connection, so scheme doesn't make sense. Related, I think we should change `url` to `host` since we are connecting to a host address via tcp, not sending a http request to a url.

**Pull request merged**
